### PR TITLE
Add parquet file with rowgroup having larger than INT16_MAX pages

### PR DESCRIPTION
**Pull request by mapleFU**
This is part of https://github.com/apache/arrow/pull/15182 and https://github.com/apache/arrow/issues/15074

This file has Schema:

```json
{
  "Version": "2.6",
  "CreatedBy": "cpp version BuildInfo:GitBranch:master,GitVersion:7def947,BuildTime:1672903249",
  "TotalRows": "40000",
  "NumberOfRowGroups": "1",
  "NumberOfRealColumns": "1",
  "NumberOfColumns": "1",
  "Columns": [
     { "Id": "0", "Name": "inc", "PhysicalType": "BOOLEAN", "ConvertedType": "NONE", "LogicalType": {"Type": "None"} }
  ],
  "RowGroups": [
     {
       "Id": "0",  "TotalBytes": "720000",  "TotalCompressedBytes": "720000",  "Rows": "40000",
       "ColumnChunks": [
          {"Id": "0", "Values": "40000", "StatsSet": "True", "Stats": {"NumNulls": "0", "DistinctValues": "0", "Max": "0", "Min": "0" },
           "Compression": "UNCOMPRESSED", "Encodings": "PLAIN", "UncompressedSize": "720000", "CompressedSize": "720000" }
        ]
     }
  ]
}
```

It has 40,000 lines, and 40000 pages for each column. The file is generated by C++ parquet.

### CODE REVIEW

**Comment by pitrou** [comment]
Hmm. I don't know if there is a way to make the file smaller? 16 MB is large for a single data file.

**Comment by mapleFU** [comment]
Sure, wait a minute, let me generate it with only one column and 10 rows per page

**Comment by mapleFU** [comment]
@pitrou I try to strip the file, but it's still huge because there are too many pages. 3.3MB now

**Comment by pitrou** [comment]
Still big, but much more reasonable, so perhaps we can live with it.

**Comment by mapleFU** [comment]
After seeing the page_index of file, the final offset indexes are:

```
              { "page_id": "49979", "offset": "1747651", "compressed_size": "35", "first_row_index": "249990" },
               { "page_id": "49980", "offset": "1747686", "compressed_size": "35", "first_row_index": "249995" }
```

So, the page data may requires 1.66MB.

The offset of footer is 455B, and it's offset is 3.1MB. So page index + offset index may occupies 1.5MB.

**Comment by wgtmac** [comment]
Can we disable writing the page index (and page statistics) and make each page have only 1 row?

**Comment by gszadovszky** [comment]
> Still big, but much more reasonable, so perhaps we can live with it. @wgtmac @gszadovszky thoughts?

I don't think 3.3MB is that big for a test file. It is another issue that we use a git repo to store these files. We might suggest in the README to not to clone the whole repo for testing but directly reference each file via github links and download only the ones are required. What do you think?

**Comment by mapleFU** [comment]
Well, can we use gzip to compress that file?

**Comment by mapleFU** [comment]
> Can we disable writing the page index (and page statistics) and make each page have only 1 row?

Only 1 row + PLAIN: 3MB
Compressed Only 1 row + PLAIN: 434KB

Maybe not writing page index can limit the page to 1.5MB? But it need to modify some code.

**Comment by wgtmac** [comment]
> > Can we disable writing the page index (and page statistics) and make each page have only 1 row?
> 
> Only 1 row + PLAIN: 3MB Compressed Only 1 row + PLAIN: 434KB
> 
> Maybe not writing page index can limit the page to 1.5MB? But it need to modify some code.

You can comment the line below to drop column index but keep offset index. I believe column index is usually larger than the offset index.

https://github.com/apache/parquet-mr/blob/433de8df33fcf31927f7b51456be9f53e64d48b9/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java#L1197

**Comment by mapleFU** [comment]
@wgtmac I use modified C++ writer to generate a file without page-index, which is 1.6MB. After gzip, it's 101kb. I think I've tried my best...
/cc @pitrou do you think we can gzip that file?

**Comment by pitrou** [comment]
> We might suggest in the README to not to clone the whole repo for testing but directly reference each file via github links and download only the ones are required. What do you think?

That would probably make the tests more fragile, so as far as Arrow C++ is concerned we'd probably stick to a git submodule.

**Comment by pitrou** [comment]
@mapleFU We should probably avoid compressing the file. The goal is ease of integration in test suites.

**Comment by mapleFU** [comment]
I make an all-nonnullable-bool and trim the page-index, which could be 700kb, do you think it's ok? @pitrou @wgtmac @gszadovszky

**Comment by pitrou** [comment]
@mapleFU Sounds great, as long as it can showcase the issue :-)

**Comment by mapleFU** [comment]
@pitrou I've upload the file and update the description here: https://github.com/apache/parquet-testing/pull/33#issue-1520221695

File: data/README.md [hunk 1]
```diff
@@ -37,6 +37,7 @@
 | datapage_v1-uncompressed-checksum.parquet      | uncompressed INT32 columns in v1 data pages with a matching CRC        |
 | datapage_v1-snappy-compressed-checksum.parquet | compressed INT32 columns in v1 data pages with a matching CRC          |
 | datapage_v1-corrupt-checksum.parquet           | uncompressed INT32 columns in v1 data pages with a mismatching CRC     |
+| overflow_i16_page_cnt.parquet                  | row group with more than INT16_MAX columns                   |
```
Comment (pitrou) [inline data/README.md:40]: I suppose it's more like this?
```suggestion
| overflow_i16_page_cnt.parquet                  | row group with more than INT16_MAX pages                   |
```

**Comment by mapleFU** [comment]
Updated

**Comment by mapleFU** [comment]
@pitrou Mind take a look? Is there some remaining issues here?

**Comment by mapleFU** [comment]
@pitrou is there any other thing I need to do on this file and description?

**Pull request merged**